diff --git a/src/art/libartbase/arch/instruction_set.h b/src/art/libartbase/arch/instruction_set.h
index 8d59f1b..cd0c1fc 100644
--- a/src/art/libartbase/arch/instruction_set.h
+++ b/src/art/libartbase/arch/instruction_set.h
@@ -24,6 +24,18 @@
 #include "base/enums.h"
 #include "base/macros.h"
 
+#define ART_STACK_OVERFLOW_GAP_arm 8192
+#define ART_STACK_OVERFLOW_GAP_arm64 16384
+#define ART_STACK_OVERFLOW_GAP_riscv64 16384
+#define ART_STACK_OVERFLOW_GAP_x86 16384
+#define ART_STACK_OVERFLOW_GAP_x86_64 20480
+
+// see aosp/art/build/art.go
+// default frame size limit: 1736
+// device limit: 7400
+// host limit: 6400
+#define ART_FRAME_SIZE_LIMIT 6400
+
 namespace art {
 
 enum class InstructionSet {