diff --git a/platform/api/task_runner.h b/platform/api/task_runner.h
index 948eb08..3673141 100644
--- a/platform/api/task_runner.h
+++ b/platform/api/task_runner.h
@@ -24,12 +24,13 @@ class TaskRunner {
  public:
 // Seem to get an error using clang when compiling with -fno-exceptions:
 //   error: implicit instantiation of undefined template 'std::__1::packaged_task<void () noexcept>'
-#if __has_feature(cxx_exceptions)
-  using Task = std::packaged_task<void() noexcept>;
-#else
-  using Task = std::packaged_task<void()>;
-#endif
+// #if __has_feature(cxx_exceptions)
+//   using Task = std::packaged_task<void() noexcept>;
+// #else
+//   using Task = std::packaged_task<void()>;
+// #endif
 
+  using Task = std::packaged_task<void()>;
   virtual ~TaskRunner() = default;
 
   // Takes any callable target (function, lambda-expression, std::bind result,
